### PR TITLE
feat : ajv to validate export conf

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
Added changes for validating export config.

### CODE REVIEW

File: packages/pods-core/src/basev3.ts [hunk 1]
```diff
@@ -225,37 +229,19 @@ export abstract class ExportPod<
...
-    const { dest } = config;
-    const configJSON = JSON.stringify(config);
-    if (_.isUndefined(dest)) {
+    const validateConfig = ajv.compile(this.config);
+    validateConfig(config);
+    if (!validateConfig(config)) {
       throw new DendronError({
-        message: `no dest specified. config: ${configJSON}`,
+        message: "validation errors",
+        payload: "error",
```
Comment (kevinslin) [inline packages/pods-core/src/basev3.ts:244]: can we return the list of errors here as well in `payload`?

File: packages/pods-core/src/builtin/AirtablePod.ts [hunk 2]
```diff
@@ -34,40 +36,36 @@ export class AirtableExportPod extends ExportPod<AirtableExportConfig> {
...
-      },
-      {
-        key: "srcFieldMapping",
-        description:
-          "mapping of airtable fields with the note eg: {Created On: created, Notes: body}",
-        type: "object",
-        required: true,
-      },
-    ]);
+    }) as JSONSchemaType<AirtableExportConfig>;
```
Comment (kevinslin) [inline packages/pods-core/src/builtin/AirtablePod.ts:68]: instead of doing a cast, can `createExportPodConfig` take a generic `T extends ExportPodConfig`?

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/pods-core/src/builtin/AirtablePod.ts (hunk 2 above)
Comment (Harshita-mindfire) [reply packages/pods-core/src/builtin/AirtablePod.ts:68]: Actually no, this results in two errors : incompatible "type" error and value of T never used.
Also, to resolve the " value of T never used" error, if we keep the return type of  createExportPodConfig to be : JSONSchemaType<T>, ajv doesn't allow a generic schema to have a type as "object".

```js
static createExportConfig<T extends ExportPodConfig>(opts: { required: string[]; properties: any }) : JSONSchemaType<T>
 {
    return {
      type: "object",
      ...
}
}
```

File: packages/pods-core/src/basev3.ts (hunk 1 above)
Comment (Harshita-mindfire) [reply packages/pods-core/src/basev3.ts:244]: Yes. Adding in message as well, since only message is visible to user.

**Review by kevinslin — APPROVED** [review]

**Pull request merged**